golem-worker-executor-base/src/durable_host/mod.rs

```diff
@@ -155,6 +158,8 @@ impl<Ctx: WorkerCtx> DurableWorkerCtx<Ctx> {
             owned_worker_id.worker_id, worker_config.deleted_regions
         );
 
+        let read_only_paths = prepare_filesystem(file_loader, temp_dir.path(), &component_metadata.files).await?;
```

Comment:
My proposal for handling files during updates replays is the following:

1. Automatic: We replay from the beginning using the new initial file set. If we encounter errors the user will have to use a snapshot based update. This is similar to how other incompatibilities in the component are being handled from what I can tell.
2. Snapshot: We rely on the user-defined save-snapshot function to save all data that the users wants to persist across updates. This could either be implemented by using the blob storage apis directly, or we could allow returning paths to files that should be persisted from the snapshot function. The user-defined load-snapshot function is then responsible to reintialize state on top of the new initial filesystem. From what I can tell this is consistent with how other state in the component is handled, so this should be least surprising to users.